proguard/src/main/java/com/flurry/proguard/UploadProguardMapping.java

```diff
@@ -0,0 +1,216 @@
+package com.flurry.proguard;
+
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.FileEntity;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.BufferedOutputStream;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.zip.GZIPOutputStream;
+
+public class UploadProguardMapping {
+
+    private static final String METADATA_BASE = "https://crash-metadata.flurry.com/pulse/v1/";
+    private static final String UPLOAD_BASE = "https://upload.flurry.com/upload/v1/upload/";
+    private static HttpClient httpClient;
+
+    public static void main(String[] args) throws IOException, InterruptedException {
+        if (args.length != 3) {
+            System.out.println("Please provide 3 arguments in the following order:");
```

Comment:
I think we should use a logging library for everywhere we have System.out.println.  In most contexts, this will be run by a build and so we should direct output that way.  That also gives us control over what gets spit out and when.